app/src/main/java/com/example/javaopencommerce/order/CardMapper.java

```diff
@@ -0,0 +1,61 @@
+package com.example.javaopencommerce.order;
+
+import static java.util.stream.Collectors.toList;
+
+import com.example.javaopencommerce.Amount;
+import com.example.javaopencommerce.Value;
+import com.example.javaopencommerce.Vat;
+import com.example.javaopencommerce.order.CardItem.CardItemSnapshot;
+import com.example.javaopencommerce.order.Item.ItemSnapshot;
+import com.example.javaopencommerce.order.dtos.CardDto;
+import com.example.javaopencommerce.order.dtos.CardItemDto;
+import com.example.javaopencommerce.order.dtos.ItemDto;
+import java.util.List;
+
+class CardMapper {
+
+  private CardMapper() {
+  }
+
+  static CardSnapshot toSnapshot(CardDto cardDto) {
+    List<CardItemSnapshot> cardItems = cardDto.getProducts().stream().map(CardMapper::toSnapshot)
+        .toList();
+
+    return new CardSnapshot(cardItems, Value.of(cardDto.getCardValueGross()),
```

Comment:
Let's have it all either in one line, or each parameter in a separate line. It's easier to read if those are either split or inline, otherwise it could be confused for two methods